clinica/utils/filemanip.py

```diff
@@ -1,209 +1,476 @@
-def zip_nii(in_file: str, same_dir: bool = False):
+from typing import List, Optional, Tuple
+
+
+def _zip_unzip_nii(in_file: str, func, same_dir):
     import gzip
+    import operator
     import shutil
     from os import getcwd
     from os.path import abspath, join
+    from pathlib import Path
 
     from nipype.utils.filemanip import split_filename
     from traits.trait_base import _Undefined
 
+    zipping = func.__name__ == "zip_nii"
```

Comment:
Could be simplified by just passing a `compress: bool` parameter instead no? Passing an entire function just to match on its name sounds like an odd pattern to me.

Reply:
Indeed, this is much better! Don't know why I did that... 🤔 
Implemented in [db32e53](https://github.com/aramis-lab/clinica/pull/781/commits/db32e53f4bd42edeae52719c3f8151cbe7b223cd)